rcljava/src/main/cpp/org_ros2_rcljava_publisher_PublisherImpl.cpp

```diff
@@ -90,3 +92,31 @@ Java_org_ros2_rcljava_publisher_PublisherImpl_nativeDispose(
     rcljava_throw_rclexception(env, ret, msg);
   }
 }
+
+JNIEXPORT jlong JNICALL
+Java_org_ros2_rcljava_publisher_PublisherImpl_nativeCreateEvent(
+  JNIEnv * env, jclass, jlong publisher_handle, jint event_type)
+{
+  auto * publisher = reinterpret_cast<rcl_publisher_t *>(publisher_handle);
+  if (!publisher) {
+    rcljava_throw_exception(
+      env, "java/lang/IllegalArgumentException", "passed rcl_publisher_t handle is NULL");
+    return 0;
+  }
+  auto * event = static_cast<rcl_event_t *>(malloc(sizeof(rcl_event_t)));
```

Comment:
I see what you mean.

The issue is that you have both `rcl_publisher_event_init` and `rcl_subscription_event_init`.
In the current approach, the publisher/subscription is responsible of creating the rcl event handle, and then the handle ownership is transferred to `EventHandlerImpl`.

The other approach is to pass a "parent type" enum to `EventHandlerImpl` together with the parent rcl handle, so it can construct both.

I'm not sure what's better, I think both approaches are a bit confusing.

Reply:
I see. Yeah, I'm not sure if passing an enum is much better.

At the very least, I think it's worth leaving a comment about the assumed transfer of ownership, probably also in the docblock for this native function.

Reply:
Fixed in https://github.com/osrf/ros2_java/pull/9/commits/60e1a001e535b89efe30c03efc670198a44378f3.